overlord/snapstate/snapmgr.go

```diff
@@ -386,6 +418,96 @@ func (m *SnapManager) blockedTask(cand *state.Task, running []*state.Task) bool
 	return false
 }
 
+var CanAutoRefresh func(st *state.State) bool
+
+// ensureRefreshes ensures that we refresh all installed snaps periodically
+func (m *SnapManager) ensureRefreshes() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// see if it even makes sense to try to refresh
+	if CanAutoRefresh == nil || !CanAutoRefresh(m.state) {
+		return nil
+	}
+
+	tr := config.NewTransaction(m.state)
+
+	// for tests
+	var refreshDisabled bool
+	err := tr.Get("core", "refresh.disabled", &refreshDisabled)
```

Comment:
Agreed, we need a decision, for now it only works when SNAPD_DEBUG is set. I could also make that SNAPPY_TESTING or even only if we have a test build (but that will be slightly problematic in the tests we run against official builds in spread-cron).